nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GetHTTP.java

```diff
@@ -467,6 +486,18 @@ public void onTrigger(final ProcessContext context, final ProcessSessionFactory
             if (accept != null) {
                 get.addHeader(HEADER_ACCEPT, accept);
             }
+
+            // Add dynamic headers
+
+            PropertyValue customHeaderValue;
+            for (PropertyDescriptor customProperty : customHeaders) {
+                customHeaderValue = context.getProperty(customProperty).evaluateAttributeExpressions();
+                if (StringUtils.isNotBlank(customHeaderValue.getValue())) {
```

Comment:
I think it might make more sense to do this in onScheduled, although depending on the decision made with the evaluation of attribute expressions, above, may be fine as it currently stands.